keps/sig-network/3458-remove-transient-node-predicates-from-service-controller/kep.yaml

```diff
@@ -10,10 +10,10 @@ reviewers:
 approvers:
   - "@thockin"
 creation-date: "2022-08-07"
-last-updated: v1.28
+last-updated: v1.29
 status: implementable
-stage: beta
-latest-milestone: "v1.28"
+stage: stable
```

Comment:
Few comment about the KEP itself:

1) In mitigations section:

> For point 3. we could implement the following change in the service controller [...]

has that happened? If not, why we believe we don't need it? can you update the KEP?

> Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested

has this happened? Can you please update the KEP with the test description and findings?


> A new metric load_balancer_sync_count will be added

Has this happened?

> A new metric nodesync_error_rate will be added

Has this happened?

Reply:
> has that happened? If not, why we believe we don't need it? can you update the KEP?

Yes. The service controller still employs the predicate which excludes nodes with the taint `ToBeDeletedByClusterAutoscaler`, see: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L1009-L1014C26

> has this happened? Can you please update the KEP with the test description and findings?

No

> Has this happened?

Yes, see: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/service/metrics.go#L44-L49

> Has this happened?

Yes, see: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/service/metrics.go#L50-L55

Reply:
For the "yes" thing - please update the KEP to reflect that (with links).


For the no thing [the test thing], can we try running something? @thockin - FYI

Reply:
> For the "yes" thing - please update the KEP to reflect that (with links). 

Done. 

> For the no thing [the test thing], can we try running something?

I am not sitting on infrastructure which allows me to create cluster from HEAD. That being said: I quite sure the upgrade path works because this has been enabled by default since 1.27 and my company has successfully upgraded clusters from 1.26 -> 1.27, with services existing on the cluster that are impacted by this change.

Reply:
I can help get access to infra to do this manual testing, if that's what we need.  That said, I'm a little skeptical that we will find anyhing by just doing it, without a) knowing what we are looking for; and b) tickling the edge cases specifically.

Reply:
OK - I buy that.

But then let's update the ` Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?` PRR question to something like

The owners of the KEP missed running the manual test when promoting the feature to Beta. Since then it was implicitly tested by many users that upgraded their clusters to 1.27+ versions without any bug reports, so running additional tests now wouldn't provide additional value.

Reply:
I've updated the answer to that question.